### PR TITLE
work on products, render products and work on single product page set authotization for seller and buyer

### PR DESCRIPTION
**Pull request by asadmalik76**
…o set authorization for seller and buyer

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -13,3 +13,241 @@
  *= require_tree .
  *= require_self
  */
+ @import url("https://fonts.googleapis.com/css2?family=Poppins:wght@100;200;300;400;500;600;700;800&display=swap");
```
Comment (ibtehaj-dnt) [inline app/assets/stylesheets/application.css:16]: use yarn package for fonts instead of this

File: app/controllers/brands_controller.rb [hunk 2]
```diff
@@ -7,14 +7,17 @@ class BrandsController < ApplicationController
 
   def index
     @brands = Brand.all
+    authorize @brands
   end
 
   def new
     @brand = Brand.new
+    authrize @brand
```
Comment (ibtehaj-dnt) [inline app/controllers/brands_controller.rb:15]: spelling mistake

File: app/controllers/brands_controller.rb [hunk 3]
```diff
@@ -7,14 +7,17 @@ class BrandsController < ApplicationController
 
   def index
     @brands = Brand.all
+    authorize @brands
```
Comment (ibtehaj-dnt) [inline app/controllers/brands_controller.rb:10]: use before action

File: app/controllers/categories_controller.rb [hunk 4]
```diff
@@ -7,14 +7,17 @@ class CategoriesController < ApplicationController
 
   def index
     @categories = Category.all
+    authorize @categories
   end
 
   def new
     @category = Category.new
+    authorize @category
```
Comment (ibtehaj-dnt) [inline app/controllers/categories_controller.rb:15]: use before action

File: app/policies/product_policy.rb [hunk 5]
```diff
@@ -9,33 +10,36 @@ def initialize(user, record)
...
-    update?
+    user.has_role? :seller and record.user == user
   end
 
   def destroy?
-    false
+    user.has_role? :seller and record.user == user
   end
 
+  # frozen_string_literal: true
```
Comment (ibtehaj-dnt) [inline app/policies/product_policy.rb:40]: you don't need to add this comment here

File: app/views/brands/edit.html.erb [hunk 6]
```diff
@@ -1,3 +1,3 @@
-<%= render "layouts/seller_menu" %>
+<%= render "partials/seller_menu" %>
```
Comment (ibtehaj-dnt) [inline app/views/brands/edit.html.erb:1]: partials is not a good name, kindly use something else

File: app/views/categories/index.html.erb [hunk 7]
```diff
@@ -20,7 +20,7 @@
       <td class="d-flex">
 
 <%= link_to "Edit", edit_category_path(category), class: "btn btn-success btn-sm" %>
-<%= button_to "Delete",  category_path(category), method: :delete, class: "btn btn-warning btn-sm"  %>
+  <%= button_to "Delete",  category_path(category), method: :delete, class: "btn btn-warning btn-sm"  %>
```
Comment (ibtehaj-dnt) [inline app/views/categories/index.html.erb:23]: please fix indentation

File: app/views/errors/unauthorized.html.erb [hunk 8]
```diff
@@ -0,0 +1,68 @@
+<html>
+
+<head>
+<style>
```
Comment (ibtehaj-dnt) [inline app/views/errors/unauthorized.html.erb:4]: kindly remove these extra tags,

File: app/views/errors/unauthorized.html.erb [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+.four_zero_four_bg{
+
+ background-image: url(https://cdn.dribbble.com/users/285475/screenshots/2083086/dribbble_1.gif);
+    height: 400px;
+    background-position: center;
+ }
+
+
+ .four_zero_four_bg h1{
```
Comment (ibtehaj-dnt) [inline app/views/errors/unauthorized.html.erb:24]: define these style in a .scss file instead

File: app/views/layouts/application.html.erb [hunk 10]
```diff
@@ -9,15 +9,18 @@
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
     <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.0.0/dist/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
+    <link rel='stylesheet' href='https://sachinchoolur.github.io/lightslider/dist/css/lightslider.css'>
```
Comment (ibtehaj-dnt) [inline app/views/layouts/application.html.erb:12]: use yarn package

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -9,15 +9,18 @@
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
     <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.0.0/dist/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
```
Comment (ibtehaj-dnt) [inline app/views/layouts/application.html.erb:11]: add bootstrap with yarn, remove all cdn

**Review by ibtehaj-dnt — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/application.css (hunk 1 above)
Comment (asadmalik76) [reply app/assets/stylesheets/application.css:16]: i was trying to use this package but unable to import all styles and js from it, will try to fix again

**Pull request merged**